integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonLoadDataCommand.scala

```diff
@@ -479,22 +479,52 @@ case class CarbonLoadDataCommand(
       LogicalRDD(attributes, rdd)(sparkSession)
 
     } else {
+      var timeStampformatString = carbonLoadModel.getTimestampformat
+      if (timeStampformatString.isEmpty) {
+        timeStampformatString = carbonLoadModel.getDefaultTimestampFormat
+      }
+      val timeStampFormat = new SimpleDateFormat(timeStampformatString)
+      var dateFormatString = carbonLoadModel.getDateFormat
+      if (dateFormatString.isEmpty) {
+        dateFormatString = carbonLoadModel.getDefaultDateFormat
+      }
+      val dateFormat = new SimpleDateFormat(dateFormatString)
       // input data from csv files. Convert to logical plan
       CommonUtil.configureCSVInputFormat(hadoopConf, carbonLoadModel)
       hadoopConf.set(FileInputFormat.INPUT_DIR, carbonLoadModel.getFactFilePath)
       val jobConf = new JobConf(hadoopConf)
       SparkHadoopUtil.get.addCredentials(jobConf)
-      val rdd = new NewHadoopRDD[NullWritable, StringArrayWritable](
-        sparkSession.sparkContext,
-        classOf[CSVInputFormat],
-        classOf[NullWritable],
-        classOf[StringArrayWritable],
-        jobConf
-      ).map(f => InternalRow.fromSeq(f._2.get().map(UTF8String.fromString)))
-
       val attributes =
         StructType(carbonLoadModel.getCsvHeaderColumns.map(
           StructField(_, StringType))).toAttributes
+      val rowDataTypes = attributes.map{f =>
+        relation.output.find(_.name.equalsIgnoreCase(f.name)) match {
+          case Some(attr) => attr.dataType
+          case _ => StringType
+        }
+      }
+      val len = rowDataTypes.length
+      val rdd =
+        new NewHadoopRDD[NullWritable, StringArrayWritable](
+          sparkSession.sparkContext,
+          classOf[CSVInputFormat],
+          classOf[NullWritable],
+          classOf[StringArrayWritable],
+          jobConf).map{f =>
```

Comment:
better to use `case (key, value)` instead of `f`

Reply:
ok